Bindings/Java/tests/TestEditMarkers.java

```diff
@@ -0,0 +1,65 @@
+import org.opensim.modeling.*;
+import java.io.IOException;
+import static java.nio.file.StandardOpenOption.*;
+import java.nio.file.*;
+
+class TestEditMarkers {
+  public static void main(String[] args) {
+    try {
+        // GUI model loading
+        Model model = new Model("gait10dof18musc_subject01.osim");
+        OpenSimContext context=null;
+        context = new OpenSimContext(model.initSystem(), model);
+        MarkerSet markerset = model.getMarkerSet();
+        // The following code exercises the Marker->Add New  functionality
+        Vec3 offset = new Vec3(0.11, 0.22, 0.33);
+        Body body = model.getBodySet().get(0);
+        String newMarkerName = "newMarker";
+        Marker newMarker = new Marker();
+        newMarker.setName(newMarkerName);
+        newMarker.set_location(offset);
+        newMarker.setParentFrameName(body.getName());
+        context.cacheModelAndState();
+        model.addMarker(newMarker);
+        try {
+            context.restoreStateFromCachedModel();
+        } catch (IOException ex) {
+            System.exit(1);
+        }
+        // This exercises Marker -> delete
+        context.cacheModelAndState();
+        
+        Marker findMarker = markerset.get(newMarkerName);
+        assert (findMarker != null);
+        markerset.remove(findMarker);
+        try {
+            context.restoreStateFromCachedModel();
+        } catch (IOException ex) {
+            System.exit(1);
+        }
+		// Exercise saveToFile
+        markerset.print("savedMarkers.xml");
+        // Now create a new MarkerSet from the file
+        MarkerSet newMarkerSet = new MarkerSet(model, "savedMarkers.xml");
+        Marker newMarkerRenamed = newMarkerSet.get(0);
+        String addedMarkerName = newMarkerRenamed.getName()+"_renamed";
```

Comment:
Just wondering; the renaming is necessary to avoid having two markers with the same name?

Reply:
Yes @chrisdembia it is and also it exercises the Marker renaming done in the GUI so that not all markers are called "NewMarker_x".